modules/xmpp/SdpConsistency.js

```diff
@@ -0,0 +1,176 @@
+import * as transform from 'sdp-transform';
+
+/**
+ * Begin helper functions
+ */
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  return the single primary video ssrcs
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  primary video ssrc
+ * @returns {number} the primary video ssrc
+ */
+function getPrimarySsrc (videoMLine) {
+    if (!videoMLine.ssrcs) {
+        return;
+    }
+    let numSsrcs = videoMLine.ssrcs
+        .filter(ssrcInfo => ssrcInfo.attribute === "msid")
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
```

Comment:
this list contains multiple entries for each ssrc (because it's each ssrc line in the sdp, so has one for msid, one for cname, etc.) so the last line ensures only unique entries are added.